src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/CLinker.java

```diff
@@ -26,23 +26,65 @@
 package jdk.incubator.foreign;
 
 import jdk.internal.foreign.NativeMemorySegmentImpl;
+import jdk.internal.foreign.PlatformLayouts;
 import jdk.internal.foreign.Utils;
 import jdk.internal.foreign.abi.SharedUtils;
 
-import java.nio.ByteOrder;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodType;
 import java.nio.charset.Charset;
 import java.util.Objects;
 import java.util.function.Consumer;
 
+import static jdk.internal.foreign.PlatformLayouts.*;
+
 /**
- * A set of utilities for working with libraries using the C language/ABI
+ * This interface is a used as a super class for all ForeinLinker implementations that
+ * implement the system's C ABI and calling convention. As such, instances of this interface
+ * can be used to link foreign functions in native libraries that follow the JVM's target
+ * platform C application binary interface (ABI).
+ *
+ * <p>There are two components that go into linking a foreign function: a method type, and
+ * a function descriptor. The method type, consisting of a set of 'carrier' types constitutes
+ * the Java side of a call to a foreign function, while the function descriptor constitutes
```

Comment:
I'd replace "constitutes the Java side of the call" with something more explicit like "...set of carrier types, which, together, specify the Java signature which clients must adhere to when e.g. calling the underlying foreign function."

Similarly "constitutes the native side of the call"; I'd replace that with "the function descriptor contains a set of memory layouts (link) which, together, uniquely specify the foreign function signature and classification information (via custom layout attributes), so that linking can take place."